pkgs/development/python-modules/tensorflow/support-numpy-1.20.patch

```diff
@@ -0,0 +1,21 @@
+diff --git a/tensorflow/python/ops/array_ops.py b/tensorflow/python/ops/array_ops.py
+index 64283ac99b0..fe5d7f20acb 100644
+--- a/tensorflow/python/ops/array_ops.py
++++ b/tensorflow/python/ops/array_ops.py
+@@ -36,6 +36,7 @@ from tensorflow.python.framework import tensor_util
+ from tensorflow.python.framework.constant_op import constant
+ from tensorflow.python.ops import gen_array_ops
+ from tensorflow.python.ops import gen_math_ops
++from tensorflow.python.ops import math_ops
+ # go/tf-wildcard-import
+ # pylint: disable=wildcard-import
+ from tensorflow.python.ops.gen_array_ops import *
+@@ -2897,7 +2898,7 @@ def matrix_set_diag(
+ 
+ def _constant_if_small(value, shape, dtype, name):
+   try:
+-    if np.prod(shape) < 1000:
++    if math_ops.reduce_prod(shape) < 1000:
+       return constant(value, shape=shape, dtype=dtype, name=name)
```

Comment:
I feel like this should probably be upstreamed, there's nothing nixpkgs specific in it.

Reply:
I agree that this fix feels like it belongs in the Tensorflow repository. (or perhaps even in the implementation of np.prod in the Numpy repository)

There have been open PRs to resolve the issue upstream/in the Tensorflow repo for more than 2 months. Supporting Numpy 1.20 doesn't seem to be a high priority for the Tensorflow team - since Numpy 1.19 is working fine with Tensorflow.

That said, Numpy 1.20-support is pretty much essential in order for Tensorflow to work within the nixpkgs ecosystem, since nixpkgs currently ships NumPy 1.20 (pkgs.python3.pkgs.numpy).

Shipping multiple versions of the same Python library within nixpkgs is a risky affair - since any environment that depends on multiple versions of the same library will break.